homeassistant/components/vicare/sensor.py

```diff
@@ -950,6 +950,8 @@ def __init__(
         """Initialize the sensor."""
         super().__init__(device_config, api, description.key)
         self.entity_description = description
+        # run update to have device_class set depending on unit_of_measurement
+        self.update()
```

Comment:
We don't allow side effects in init methods. This needs to be refactored. Instead pass `True` as second argument to `async_add_entities`.